torchgeo/datasets/millionaid.py

```diff
@@ -0,0 +1,397 @@
+    splits = ["train", "test"]
+
+    def __init__(
+        self,
+        root: str = "data",
+        task: str = "multi-class",
+        split: str = "train",
+        transforms: Optional[Callable[[Dict[str, Tensor]], Dict[str, Tensor]]] = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new MillionAID dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            task: whether to use multi-class or multi-label task
+            split: train or test split
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            RuntimeError: if ``download=False`` and data is not found, or checksums
+                don't match
+        """
+        self.root = root
+        self.transforms = transforms
+        self.download = download
+        self.checksum = checksum
+        assert task in self.tasks
+        assert split in self.splits
+        self.task = task
+        self.split = split
+
+        self._verify()
+
+        self.files = self._load_files(self.root)
+
+        self.classes = sorted(set(cls for f in self.files for cls in f["label"]))
+        self.class_to_idx: Dict[str, int] = {c: i for i, c in enumerate(self.classes)}
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.files)
+
+    def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        files = self.files[index]
+        image = self._load_image(files["image"])
+        cls_label = [self.class_to_idx[label] for label in files["label"]]
+        label = torch.tensor(cls_label, dtype=torch.long)  # type: ignore[attr-defined]
+        sample = {"image": image, "label": label}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def _load_files(self, root: str) -> List[Dict[str, Any]]:
+        """Return the paths of the files in the dataset.
+
+        Args:
+            root: root dir of dataset
+
+        Returns:
+            list of dicts containing paths for each pair of image, and list of labels
+        """
+        imgs_no_subcat = list(
+            glob.glob(os.path.join(root, self.split, "**", "**", "*.jpg"))
+        )
+
+        imgs_subcat = list(
+            glob.glob(os.path.join(root, self.split, "**", "**", "**", "*.jpg"))
+        )
+
+        scenes = [p.split("/")[-3] for p in imgs_no_subcat] + [
```

Comment:
```suggestion
        scenes = [p.split(os.sep)[-3] for p in imgs_no_subcat] + [
```
Same for other instances. This will fix the failing Windows tests.